docs/book/src/developer/providers/contracts/infra-machinepool.md

```diff
@@ -0,0 +1,525 @@
+}
+```
+
+By opting into MachinePool Machines its the responsibility of the provider to create an instance of a InfraMachine for every replica and manage their lifecycle.
+
+### InfraMachinePool: instances
+
+Each InfraMachinePool MAY specify a status field that is used to report information about each instance within the machine pool. This field is purely informational and is used as convenient way for a user to get details of the instances such as provider id and addresses.
+
+If you implement this then create a `status.instances` field that is a slice of a struct type that contains the information you want to store and be made available to the users.
+
+```go
+type FooMachinePoolStatus struct {
+    // Instances contains the status for each instance in the pool
+    // +optional
+    Instances []FooMachinePoolInstanceStatus `json:"instances,omitempty"`
+    
+    // See other rules for more details about mandatory/optional fields in InfraMachinePool status.
+    // Other fields SHOULD be added based on the needs of your provider.
+}
+
+// FooMachinePoolInstanceStatus contains instance status information about a FooMachinePool.
+type FooMachinePoolInstanceStatus struct {
+ // Addresses contains the associated addresses for the machine.
+ // +optional
+ Addresses []clusterv1.MachineAddress `json:"addresses,omitempty"`
+
+ // InstanceName is the identification of the Machine Instance within the Machine Pool
+ InstanceName string `json:"instanceName,omitempty"`
+
+ // ProviderID is the provider identification of the Machine Pool Instance
+ // +optional
+ ProviderID *string `json:"providerID,omitempty"`
+
+ // Version defines the Kubernetes version for the Machine Instance
+ // +optional
+ Version *string `json:"version,omitempty"`
+
+ // Ready denotes that the machine is ready
+ // +optional
+ Ready bool `json:"ready"`
+}
+```
+
+### InfraMachinePool: providerID
+
+Each InfraMachinePool MAY specify a provider ID on `spec.providerID` that can be used to identify the InfraMachinePool.
+
+```go
+type FooMachinePoolSpec struct {
+    // providerID is the identification ID of the FooMachinePool.
+    // +optional
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=512
+	ProviderID string `json:"providerID,omitempty"`
+    
+    // See other rules for more details about mandatory/optional fields in InfraMachinePool spec.
+    // Other fields SHOULD be added based on the needs of your provider.
+}
+```
+
+NOTE: To align with API conventions, we recommend since the v1beta2 contract that the `ProviderID` field should be
+of type `string`.
+
+
+### InfraMachinePool: providerIDList
+
+Each InfraMachinePool MUST supply a list of the identification IDs of the machine instances managed by the machine pool by storing these in `spec.providerIDList`.
+
+```go
+type FooMachinePoolSpec struct {
+    // ProviderIDList is the list of identification IDs of machine instances managed by this Machine Pool
+	// +optional
```

Comment:
Let's align the field definition to what we have in CAPI MachinePool

```suggestion
	// +optional
	// +listType=atomic
	// +kubebuilder:validation:MaxItems=10000
	// +kubebuilder:validation:items:MinLength=1
	// +kubebuilder:validation:items:MaxLength=512
```

(please also double check other fields, we invested a lot in the latest API iteration on getting markers right)